src/lib/core/usecase/case-study-usecase.ts

```diff
@@ -0,0 +1,395 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { type RemoteFile } from "../entity/file";
+import { type CaseStudyInputPort, type CaseStudyOutputPort } from "../ports/primary/case-study-primary-ports";
+import type AgentGatewayOutputPort from "../ports/secondary/agent-gateway-output-port";
+import type CaseStudyRepositoryOutputPort from "../ports/secondary/case-study-repository-output-port";
+import type ResearchContextGatewayOutputPort from "../ports/secondary/research-context-gateway-output-port";
+import type SourceDataGatewayOutputPort from "../ports/secondary/source-data-gateway-output-port";
+import type VectorStoreOutputPort from "../ports/secondary/vector-store-output-port";
+import { type TCaseStudyRequest } from "../usecase-models/case-study-usecase-models";
+
+export default class BrowserCaseStudyUsecase implements CaseStudyInputPort {
+  presenter: CaseStudyOutputPort;
+  researchContextGateway: ResearchContextGatewayOutputPort;
+  agentGateway: AgentGatewayOutputPort<any>;
+  vectorStore: VectorStoreOutputPort;
+  sourceDataGateway: SourceDataGatewayOutputPort;
+  caseStudyRepository: CaseStudyRepositoryOutputPort;
+  constructor(
+    presenter: CaseStudyOutputPort,
+    researchContextsGateway: ResearchContextGatewayOutputPort,
+    agentGateway: AgentGatewayOutputPort<any>,
+    vectorStore: VectorStoreOutputPort,
+    sourceDataGateway: SourceDataGatewayOutputPort,
+    caseStudyRepository: CaseStudyRepositoryOutputPort,
+  ) {
+    this.presenter = presenter;
+    this.researchContextGateway = researchContextsGateway;
+    this.agentGateway = agentGateway;
+    this.vectorStore = vectorStore;
+    this.sourceDataGateway = sourceDataGateway;
+    this.caseStudyRepository = caseStudyRepository;
+  }
+
+  async execute(request: TCaseStudyRequest): Promise<void> {
```

Comment:
@maany Most important part of this PR is in this function, and the new SDACaseStudyRepository. The rest is  just scaffolding/wiring

Reply:
reviewed the usecase. Covers quite some edge cases that I could think of. Nice work